firestore/query.go

```diff
@@ -103,7 +122,12 @@ func (q Query) Where(path, op string, value interface{}) Query {
 		q.err = err
 		return q
 	}
-	q.filters = append(append([]filter(nil), q.filters...), filter{fp, op, value})
+	proto, err := filter{fp, op, value}.toProto()
+	if err != nil {
+		q.err = err
+		return q
+	}
+	q.filters = append(append([]*pb.StructuredQuery_Filter(nil), q.filters...), proto)
```

Comment:
nit: Can this be two separate lines, then no initialization is needed.

Reply:
offline chat. this is to result in a copied filter slice.